From 342a3f02d92421d6cc8d6b5f41db672396b94671 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Thu, 11 Aug 2016 03:50:36 +0200
Subject: [PATCH] Resolve unit test failures on arm and mips

Printing guint64 values doesn't work the same on all platforms and was
wrongly implemented.
Good that we have unit tests!
This should resolve the issue in a (mostly) portable way.
---
 src/as-xmldata.c  | 4 ++--
 src/as-yamldata.c | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/as-xmldata.c b/src/as-xmldata.c
index 465aed5..2ad2929 100644
--- a/src/as-xmldata.c
+++ b/src/as-xmldata.c
@@ -600,7 +600,7 @@ as_xmldata_process_releases_tag (AsXMLData *xdt, xmlNode *node, AsComponent *cpt
 
 			prop = (gchar*) xmlGetProp (iter, (xmlChar*) "timestamp");
 			if (prop != NULL) {
-				timestamp = g_ascii_strtoll (prop, NULL, 10);
+				timestamp = atol (prop);
 				as_release_set_timestamp (release, timestamp);
 				g_free (prop);
 			}
@@ -1396,7 +1396,7 @@ as_xmldata_add_release_subnodes (AsXMLData *xdt, AsComponent *cpt, xmlNode *root
 			g_autofree gchar *time_str = NULL;
 
 			if (priv->mode == AS_PARSER_MODE_DISTRO) {
-				time_str = g_strdup_printf ("%ld", unixtime);
+				time_str = g_strdup_printf ("%" G_GUINT64_FORMAT, unixtime);
 				xmlNewProp (subnode, (xmlChar*) "timestamp",
 						(xmlChar*) time_str);
 			} else {
diff --git a/src/as-yamldata.c b/src/as-yamldata.c
index 91b54f9..847dbae 100644
--- a/src/as-yamldata.c
+++ b/src/as-yamldata.c
@@ -679,7 +679,7 @@ as_yamldata_process_releases (AsYAMLData *ydt, GNode *node, AsComponent *cpt)
 			value = as_yaml_node_get_value (n);
 
 			if (g_strcmp0 (key, "unix-timestamp") == 0) {
-				as_release_set_timestamp (rel, g_ascii_strtoll (value, NULL, 10));
+				as_release_set_timestamp (rel, atol (value));
 			} else if (g_strcmp0 (key, "date") == 0) {
 				g_autoptr(GDateTime) time;
 				time = as_iso8601_to_datetime (value);
@@ -1639,7 +1639,7 @@ as_yaml_data_emit_releases (AsYAMLData *ydt, yaml_emitter_t *emitter, AsComponen
 			g_autofree gchar *time_str = NULL;
 
 			if (priv->mode == AS_PARSER_MODE_DISTRO) {
-				time_str = g_strdup_printf ("%ld", unixtime);
+				time_str = g_strdup_printf ("%" G_GUINT64_FORMAT, unixtime);
 				as_yaml_emit_entry (emitter, "unix-timestamp", time_str);
 			} else {
 				GTimeVal time;
